new_lessons/sets.md

```diff
@@ -0,0 +1,337 @@
+# Sets 
+
+## Introduction 
+
+Sets are a unique data structure in Python.  Sets can be used to store data like lists and dictionaries, but sets also have a wide array of built in functionality based on the concept of sets in mathematics, and this is where sets really shine!  In this lesson we will explore the basics of sets and look at the ways we can take advantage of the features of sets in algorithms and problem solving.
+
+## Vocabulary and Synonyms 
+
+| Vocab | Definition | Synonyms | How to Use in a Sentence |
+| ----------------- | ----------------------------------------------------------------------------------------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Set |  |  |
+| Union |  |  |
+| Intersection |  |  |
+| Difference |  |  |
+| Disjoint |  |  |
+
+## Set Basics
+
+Sets are like buckets that can hold things!  Just like a bucket, we can add things to a set, remove things from a set, and dump everything out.  In addition, we can do many interesting things around comparing the contents of one set to another and combining the contents in specific ways.
+
+ There are a few rules that govern sets:
+
+1. Each element of a set is unique.
+1. Each element of a set must be immutable.
+1. Elements in a set are unordered.
+
+### Set Operations and Syntax
+
+1. Create a new set:
+```python
+
+new_set = set() # To create an empty set, use the keyword set()
+set_with_contents = {1, 2, 3, 'a', 'b'} # To create a set with contents, use {}
+# like dictionaries.  Notice the difference from dictionaries, each element is
+# on it's own, there are no key-value pairs.
+
+```
```

Comment:
Overall throughout this lesson, there's inconsistency between if code snippets have empty lines on top and bottom, only on bottom, only on top, or none.

Could we conform to one?

The rest of the curriculum does not have empty lines on the top or bottom.